src/recommendationservice_ballerina/src/recommendationservice/RecommendationService_sample_service.bal

```diff
@@ -1,13 +1,70 @@
 import ballerina/grpc;
+import ballerina/io;
+import ballerina/kubernetes;
 
-listener grpc:Listener ep = new (9090);
+// Product catalog client.
+ProductCatalogServiceBlockingClient productCat = new ("http://productcatalogservice:3550");
 
-service RecommendationService on ep {
+@kubernetes:Service {
+    serviceType: "ClusterIP",
+    name: "recommendationservice"
+}
+service RecommendationService on new grpc:Listener(8080) {
 
     resource function ListRecommendations(grpc:Caller caller, ListRecommendationsRequest value) {
-        // Implementation goes here.
+        Empty req = {};
+        // Fetch list of products from product catalog stub
+        var products = productCat->ListProducts(req);
+        if (products is grpc:Error) {
+            io:println("Error from Connector: " + products.reason() + " - "
+            + <string>products.detail()["message"]);
+
+            // You should return a ListRecommendationsResponse
+            ListRecommendationsRequest resp = {
+                user_id: value.user_id,
+                product_ids: ["9SIQT8TOJO", "6E92ZMYYFZ", "LS4PSXUNUM"]
+            };
+            io:println(resp);
+            var e = caller->send(resp);
+            e = caller->complete();
+        } else {
+            ListProductsResponse listProductResponse;
+            grpc:Headers headers;
+            [listProductResponse, headers] = products;
```

Comment:
L31-L33 can be merged

```suggestion
            [ListProductsResponse, grpc:Headers] [listProductResponse, headers] = products;
```

Reply:
If we are not using `headers` we can just do

```
ListProductsResponse listProductResponse = products[0];
```

Reply:
this looks great ... will add this